test/deriving_warning/run.t

```diff
@@ -235,3 +238,55 @@ Same goes for .mli:
   include sig [@@@ocaml.warning "-32"] val unit_two : unit end[@@ocaml.doc
                                                                 "@inline"]
   [@@merlin.hide ]
+
+--------------------------------------------------------------------------------
+
+Whenever a set of types has a [@@deriving ...] attached, ppxlib's driver always
+generate structure items meant to disable unused type warnings (warning 34) for
+any of those types.
+
+Let's consider the following piece of OCaml code:
+
+  $ cat > unused_types.ml << EOF
+  > type t = int
+  > and u = string
+  > [@@deriving zero_do_warn]
+  > EOF
+
+If we run the driver:
+
+  $ ./driver.exe -impl unused_types.ml
+  type t = int
+  and u = string[@@deriving zero_do_warn]
+  include struct let _ = fun (_ : t) -> ()
+                 let _ = fun (_ : u) -> () end[@@ocaml.doc "@inline"][@@merlin.hide
+                                                                      ]
+  include
+    struct
+      [@@@ocaml.warning "-60"]
+      module Zero = struct type t =
+                             | T0  end
+      let zero = Zero.T0
+      let _ = zero
+    end[@@ocaml.doc "@inline"][@@merlin.hide ]
+
+We can see that the driver generated two `let _ = fun (_ : ...`, one for each type
+in the set.
+
+We have a flag that disable this behaviour and allows unused type warnings to be
+reported properly. Passing that flag to the driver should remove the two previously
+mentionned items.
```

Comment:
```suggestion
mentioned items.
```

Reply:
@patricoferris I will verify how much of the original code I re-used and apply your review comments to the other branch as well. Thanks!